src/com/company/Main.java

```diff
@@ -1,40 +1,62 @@
 package com.company;
 
 import java.io.File;
-import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.sql.*;
 import java.util.StringTokenizer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class Main {
     final static String FILE_TYPE_TXT = ".txt";
     final static String FILE_TYPE_JSON = ".json";
-    final static String SPECIAL_CHAR_REMOVAL_REGEX = "[^a-zA-Z0-9@]";
+    final static String SPECIAL_CHAR_REMOVAL_REGEX = "[^a-zA-Z0-9]";
     final static String SINGLE_SPACE = " ";
+    static PreparedStatement statement;
+    static String txtFilePath = null;
+    static String searchWord = null;
+    static int count = 0;
+    static String status = "Failure";
+    static Database database = new Database();
+
+    public static void main(String[] args) throws SQLException {
 
-    public static void main(String[] args) {
 
         if (args.length != 2) {
-            System.out.println("Exiting, since no file path and word to search for are provided!");
+            String errorMessage = "Exiting, since no file path and word to search for are provided!";
+            System.out.println(errorMessage);
+            database.databaseInsertion(txtFilePath, searchWord, count, status, errorMessage);
             return;
         }
-        String txtFilePath = args[0];
-        String searchWord = args[1];
-        File file = new File(txtFilePath);
+        txtFilePath = args[0];
+        searchWord = args[1];
 
-        if (!file.exists() || !isFileFormatSupported(file)) {
-            System.out.println("File is not valid");
+        Pattern pattern = Pattern.compile(SPECIAL_CHAR_REMOVAL_REGEX);
+        Matcher matcher = pattern.matcher(searchWord);
+        boolean isStringContainsSpecialCharacter = matcher.find();
+        if (isStringContainsSpecialCharacter) {
+            String errorMessage = "the search Word contains special character";
+            System.out.println(errorMessage);
+            database.databaseInsertion(txtFilePath, searchWord, count, status, errorMessage);
             return;
         }
-        if (file.length() == 0) {
-            System.out.println("The file does not contains any data ");
+
+        File file = new File(txtFilePath);
+
+        if (!file.exists() || !isFileFormatSupported(file) || file.length() == 0) {
+            String errorMessage = "File is not valid or file doesn't exist or File doesn't contain any data";
+            System.out.println(errorMessage);
+            database.databaseInsertion(txtFilePath, searchWord, count, status, errorMessage);
             return;
         }
+
         System.out.println("Processing................");
         String data = readFileAsString(txtFilePath);
-        if(data==null)
-        {
+        if (data == null) {
             System.out.println("Couldn't read data from the file");
+            String errorMessage = "Couldn't read data from the file";
```

Comment:
Couldn't read data from the file string is used in line 57 and 58, so make the text a static final constant

Reply:
yeah changed in to final constant.